libs/designsystem/calendar/src/calendar.component.ts

```diff
@@ -351,67 +354,57 @@ export class CalendarComponent implements OnInit, AfterViewInit, OnChanges {
 
-      const isSelectable = this.isSelectable(day, cellDate);
-      const isSelected = isSameDay(this.selectedDate, cellDate);
       const cell: CalendarCell = {
         date: cellDate.getDate(),
         monthIndex: cellDate.getMonth(),
         year: cellDate.getFullYear(),
-        isCurrentMonth: day.isCurrentMonth,
-        isSelectable,
-        isSelected,
-        cssClasses: this.getCssClasses(day, isSelectable, isSelected),
+        ariaLabel: this.formatWithLocale(cellDate, this.formatDateLabel()),
+        ...this.getCalendarCellMetadata(cellDate, today, monthStart),
       };
-      if (isSelected) {
+      if (cell.isSelected) {
         this.selectedDay = cell;
       }
+      if (cell.isFocussed) {
+        this.focussedDay = cell;
+      }
       return cell;
     });
     this._month = this.chunk(days, 7);
   }
 
-  private getCalendarDay(date: Date, today: Date, monthStart: Date): CalendarDay {
+  private getCalendarCellMetadata(date: Date, today: Date, monthStart: Date): CalendarCellMetadata {
     return {
       isToday: isSameDay(today, date),
       isPast: isBefore(date, today),
       isFuture: isAfter(date, today),
       isWeekend: isWeekend(date),
       isCurrentMonth: isSameMonth(date, monthStart),
-      isDisabled: this.isDisabledDate(date) || !this.isEnabledDate(date),
+      isSelectable: this.isSelectable(date, today),
+      isFocusable: this.isFocusable(date, today),
+      isSelected: isSameDay(this.selectedDate, date),
+      isFocussed: isSameDay(this.focussedDate, date),
     };
   }
 
-  private isSelectable(day: CalendarDay, date: Date) {
-    return (
-      (this.alwaysEnableToday && day.isToday) ||
-      (!day.isDisabled &&
-        !(this.disableWeekends && day.isWeekend) &&
-        !(this.disablePastDates && day.isPast) &&
-        !(this.disableFutureDates && day.isFuture) &&
-        !(this.minDate && isBefore(date, this.minDate)) &&
-        !(this.maxDate && isAfter(date, this.maxDate)))
-    );
+  private isSelectable(date: Date, today: Date): boolean {
+    if (this.alwaysEnableToday && isSameDay(today, date)) return true;
+
+    if (this.isDisabledDate(date)) return false;
+    if (!this.isEnabledDate(date)) return false;
+    if (this.disableWeekends && isWeekend(date)) return false;
+    if (!this.isFocusable(date, today)) return false;
+
+    return true;
   }
 
-  private getCssClasses(day: CalendarDay, isSelectable: boolean, isSelected: boolean) {
-    const cssClasses = {
-      'current-month': day.isCurrentMonth,
-      weekend: day.isWeekend,
-      today: day.isToday,
-      selectable: isSelectable,
-      selected: isSelected,
-      past: day.isPast,
-      disabled: day.isDisabled,
-    };
-    let cssClassString = 'day';
-    for (const key in cssClasses) {
-      if (cssClasses[key]) {
-        cssClassString += ' ' + key;
-      }
-    }
-    return cssClassString;
+  private isFocusable(date: Date, today: Date) {
```

Comment:
I see we reuse this in the `isSelectable` method, which made me wonder "why does the ability to focus control whether it's selectable"? 🤔 
So maybe rename or extract to new method:
```suggestion
  private isWithinAllowedRange(date: Date, today: Date) {
```